### PR TITLE
Hisui

### PR DESCRIPTION
**Pull request by anhthang**
I have added some new data for the new Pokemon and new forms introduced in Legends: Arceus

All of this data was collected manually from Bulbapedia (Japanese collected from Zukan)

Not sure what I'm doing is in the right way, or missing anything, so please help me verify if possible, then I can continue and correct it.

### CODE REVIEW

**Comment by Naramsim** [comment]
Hi @anhthang, sorry for the extremely late reply. Thanks infinitely for the additions. I'll have to review a bit the PR and understand how the new update works.

If I understood correctly this data is from the Arceus update, right? So we are still talking about gen 8?

**Comment by anhthang** [comment]
Hi @Naramsim 

Yes, this is gen 8 update from Arceus. So, please review it.

I also find out that some old pokemon has forms (cherrim, shellos, gastrodon, deerling, sawsbuck...), but they're missing pokemon/species id so it's missing in the API too. How we can add it safely? I thought it should be in a order like pokemon_id, is it?

**Comment by simonorono** [comment]
This is great @anhthang  I noticed that the new Pokédex from Hisui, the addition of new Pokémon to the national dex and new moves/learn-sets is missing. Would that block the merging of this PR or can be contributed later? @Naramsim

I could help with those.

**Comment by anhthang** [comment]
Thanks @simonorono. I'm still waiting feedback from @Naramsim before adding the others. New moves, TM/HM... maybe in another PR.

**Comment by Naramsim** [comment]
I don't have any time to look into it guys. I'm sorry this has taken ages. Hopefully, I'll reserve some time and try to understand the new generation and the new data. But if there are other core members who are willing to review, then it might be way quicker.

**Comment by Naramsim** [comment]
If you want we could test this data in our staging environment. Could you open an identical PR towards the `staging` branch of PokeAPI?

File: pokedex/data/csv/versions.csv [hunk 1]
```diff
@@ -33,3 +33,4 @@ id,version_group_id,identifier
 32,19,lets-go-eevee
 33,20,sword
 34,20,shield
+35,20,legends-arceus
```
Comment (Naramsim) [inline pokedex/data/csv/versions.csv]: Hi! Sorry for coming back only now. I think the PR looks good, but I'd like to request a change/addition. 

In between _shield_ and _legends-arceus_ there are other 4 versions that we didn't add. We added the pokemon released in those 4 versions but we didn't actually add the versions.

Could you insert the versions:
- The Isle of Armor
- The Crown Tundra
- Brilliant Diamond
- Shining Pearl

And the version groups: 
- The Isle of Armor
- The Crown Tundra
- Brilliant Diamond and Shining Pearl

Thanks so much! I think you'll need also to update the other files you contributed to, such as `pokemon_forms.csv`

**Review by Naramsim — CHANGES_REQUESTED** [review]

File: pokedex/data/csv/versions.csv (hunk 1 above)
Comment (anhthang) [reply pokedex/data/csv/versions.csv]: Sure, I will check it and update.

Is that 4 version are same group? How about the Legends Arceus, same or another one?

Comment (Naramsim) [reply pokedex/data/csv/versions.csv]: So, there are 4 new versions-groups to be added:

```
the-isle-of-armor
the-crown-tundra
brilliant-diamond-and-shining-pearl
legends-arceus
```

and there are 5 new versions:

```
the-isle-of-armor
the-crown-tundra
brilliant-diamond
shining-pearl
legends-arceus
```

**Comment by anhthang** [comment]
Hi @Naramsim,

Versions & version groups are updated, pls verify again if you have time.

I also have a question, why some pokemon has forms but they don't have a pokemon_id (like Cherrim, Unown...)

**Review by Naramsim — APPROVED** [review]

**Comment by Naramsim** [comment]
Hi @anhthang , thanks so much again! Imho, we can merge. @C-Garza any thoughts?

About the forms: there are forms and varieties. I forgot where the issue describing them is. I should pin it when I find it again.

**Comment by C-Garza** [comment]
Sorry, I haven't had time to look at this. I actually didn't notice this PR until you pinged me 😅. I can probably take a look at it tomorrow if you haven't already merged it by then.

**Comment by anhthang** [comment]
just fixed the flavor texts, break lines and sort

File: pokedex/data/csv/pokemon_species_flavor_text.csv [hunk 2]
```diff
@@ -123106,6 +123133,12 @@ a deity in Sinnoh-region mythology."
 484,26,11,"空間を　ゆがめる　能力を　持ち
 シンオウ地方の　神話では
 神様として　描かれている。"
+484,39,9,"This Pokémon is feared as a deity in Hisuian legend. The birth
+of Palkia was what caused the walls of our world to disappear,
+creating a sky that spans for infinity."
+484,39,9,"It soars across the sky in a form that greatly resembles the
+creator of all things. Perhaps this imitation of appearance is
+Palkia's strategy for gaining Arceus's powers."
```
Comment (C-Garza) [inline pokedex/data/csv/pokemon_species_flavor_text.csv]: I see two flavor texts for the same entry. I'm assuming this pokemon has an additional entry for a form or something. I remember there being an issue about this before and the way it was resolved was for now was removing the form's entry. Same thing for id 483 and 905.
[Found the old issue](https://github.com/PokeAPI/pokeapi/issues/533)

**Comment by C-Garza** [comment]
I left one comment, other than that it looks good to me.

File: pokedex/data/csv/pokemon_species_flavor_text.csv (hunk 2 above)
Comment (anhthang) [reply pokedex/data/csv/pokemon_species_flavor_text.csv]: Thanks for addressing that. I have removed just now.

**Comment by Naramsim** [comment]
Nice, I'll try build pokeapi with this data later

File: pokedex/data/csv/pokemon_species.csv [hunk 3]
```diff
@@ -897,3 +897,10 @@ id,identifier,generation_id,evolves_from_species_id,evolution_chain_id,color_id,
 896,glastrier,8,,474,9,8,,-1,3,35,0,120,0,1,0,1,0,896,
 897,spectrier,8,,475,1,8,,-1,3,35,0,120,0,1,0,1,0,897,
 898,calyrex,8,,476,5,12,,-1,3,100,0,120,0,1,0,1,0,898,
+899,wyrdeer,8,234,,4,,,4,,,0,,,1,,0,0,899,
```
Comment (Naramsim) [inline pokedex/data/csv/pokemon_species.csv]: Hi, here the data lacks information. By building the DB I get that:

> null value in column "forms_switchable" of relation "pokemon_v2_pokemonspecies" violates not-null constraint

I guess we should try to fill all the columns.

Comment (Naramsim) [reply pokedex/data/csv/pokemon_species.csv]: Here the build: https://app.circleci.com/pipelines/github/PokeAPI/pokeapi/511/workflows/19090bb7-cda9-40cb-8187-9002a65fca2a/jobs/1228

Comment (anhthang) [reply pokedex/data/csv/pokemon_species.csv]: @Naramsim updated in last commit

**Review by Naramsim — COMMENTED** [review]
Nice thanks!

File: pokedex/data/csv/pokemon_species_flavor_text.csv [hunk 4]
```diff
@@ -63751,6 +63940,9 @@ many people detest it."
 いわれ　忌み嫌う　人も　多い。"
 198,30,12,"会为老大去寻找闪闪发光的东西。
 据说它很不吉利，厌恶它的人也很多。"
+198,39,9,"Widely shunned as a bearer of ill fortune. Upon crossing paths
+with this creature, I've been told one must chant "Workrum,
```
Comment (Naramsim) [inline pokedex/data/csv/pokemon_species_flavor_text.csv:63944]: Here the build complained about the use of `"`. They shouldn't be there. Maybe you can replace the double quotes with single ones.

Comment (anhthang) [reply pokedex/data/csv/pokemon_species_flavor_text.csv:63944]: @Naramsim I have fixed, can you rebuild again?

Comment (Naramsim) [reply pokedex/data/csv/pokemon_species_flavor_text.csv:63944]: I'll try!

Comment (Naramsim) [reply pokedex/data/csv/pokemon_species_flavor_text.csv:63944]:
```
building pokemon_v2_pokemonform
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 353, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/commands/shell.py", line 92, in handle
    exec(sys.stdin.read())
  File "<string>", line 1, in <module>
  File "/code/data/v2/build.py", line 2272, in build_all
    _build_pokemons()
  File "/code/data/v2/build.py", line 1965, in _build_pokemons
    build_generic((PokemonForm,), "pokemon_forms.csv", csv_record_to_objects)
  File "/code/data/v2/build.py", line 92, in build_generic
    for obj in csv_record_to_objects(csv_record):
  File "/code/data/v2/build.py", line 1962, in csv_record_to_objects
    order=int(info[9]),
ValueError: invalid literal for int() with base 10: ''
```

The order has to be set, I guess. I remember that one could put 0.

Comment (anhthang) [reply pokedex/data/csv/pokemon_species_flavor_text.csv:63944]: hi @Naramsim, I just added the order, it's increased from the last number

Comment (Naramsim) [reply pokedex/data/csv/pokemon_species_flavor_text.csv:63944]: Hi, you forgot to add the order to IDs 899->905.

Maybe you can add it, and shift the other `order`s you added previously

Comment (anhthang) [reply pokedex/data/csv/pokemon_species_flavor_text.csv:63944]: Just added @Naramsim

**Comment by Naramsim** [comment]
Yay! It's working!

**Comment by Naramsim** [comment]
Can you test a bit? https://staging.pokeapi.co/api/v2/pokemon/?offset=0&limit=200000 @anhthang

**Comment by anhthang** [comment]
@Naramsim I just fixed a small bug. I think the other is fine to me

**Comment by Naramsim** [comment]
@anhthang, it's live! Thank you so much for the effort in these months. I know I took quite a while, but we maintainers don't have much time anymore.

Thanks again!

**Comment by anhthang** [comment]
@Naramsim thank you for your support 💪 can you update graphql also? it seems not updated

**Comment by Naramsim** [comment]
Ahhh, hahah yeahh, I'll try!!!

**Comment by Naramsim** [comment]
I've updated the graphql, can you check?

**Comment by anhthang** [comment]
> I've updated the graphql, can you check?

Just tested, it seems working fine except some optional data. I will work on it later if I have time

**Comment by hexbyt3** [comment]
Was this ever merged and fixed?  All of the Hisui mons don't have gen 8 learnsets or moves that I can pull.

**Comment by Naramsim** [comment]
Yes this was merged

**Pull request merged**